### DISCUSSION TITLE
Ring-Mod Side-Chain mix idea for next plugin

### DISCUSSION THREAD

**Discussion by KlashKontakt** [ryukau/VSTPlugins#30] [Ideas]
Hello!

There is some idea not mine but in general has not yet been implemented by anyone in the form of a separate plugin.

https://youtu.be/OzLEwdcDXWo

I think it would be a very useful tool. What do you think about it?

Thnx for Your Time!

Cheers 

Alexey!

**Comment by ryukau** [comment]
I watched the video, and my understanding is shown in a block diagram below.

![ringmod_sidechain](https://user-images.githubusercontent.com/6243462/223475499-8892ce39-e790-4a79-82da-9ed39d7f29b7.png)

Let me know if above is correct or not. Also if possible, please provide a block diagram or a signal flow diagram like above. It would greatly speed up the discussion.

I'll experiment a bit about the idea. But honestly, it feels like preparation of source materials matters more than ring modulation.

**Comment by KlashKontakt** [comment]
This not my project but I hope its help for understanding 

https://disk.yandex.ru/d/jqzc3Rlf0zLm8A

https://forum.renoise.com/t/ringmod-sidechain/68071

**Comment by ryukau** [comment]
Okay, I read the Renoise forum link. It sounds like a mixing technique rather than adding effects.

How to handle a case, where side chain signal goes over 0 dB, might be a problem. I'll probably just clip the signal, but will think a bit about it.

**Comment by ryukau** [comment]
The prototype ImmediateSpacer is available on the link below. The link is the same one on [this comment](https://github.com/ryukau/VSTPlugins/discussions/29#discussioncomment-5241930).

- https://drive.google.com/file/d/1lli-76aEo8_e8GJm9vFW0ru0su7ZN680/view?usp=sharing

Let me know if it works as intended. Note that range and sensitivity of gains are not well tuned.

**Comment by ryukau** [comment]
I personally consider this as a modulator rather than mixing tool. This is because the distortion caused by ring modulation.

The idea of this ring mod mixing is similar to how digital limiter works. For smoother sounds, I think it's better to use side chain limiter.

**Comment by KlashKontakt** [comment]
Спасибо. Вчера я тестировал этот инструмент весь вечер, и все, кажется, работает, но для того, чтобы получить эффект ограничения определенной группы звуков, вам нужно отключить отправку из общего микширования и оставить только сигнал суммы в плагине. Я предположил, что сигнал, к которому применяется обработанный, будет вычтен из обработанного сигнала. Или я перехитрил себя? но в целом, если мы говорим о звуковом дизайне, то такая полезность - это довольно крутая вещь в контексте моих звуковых исследований. Я имею в виду способность вычитать из основного сигнала, смешанного из боковой цепи. Я имею в виду не динамически, а частоту или спектрально


as an example, a drum loop subtracts from pure noise.

**Comment by ryukau** [comment]
Just in case, I can only read English or Japanese. Maybe you forget to translate after edit? The notification email from GitHub was translated in English. Copy of the translation is on below.

---

Translation:

Thank you.  yesterday I tested this instrument all evening and everything seems to work, but in order to get the effect of limiting a specific group of sounds, you have to turn off the send from the general mix and leave only the sum signal in the plugin.  I assumed that the signal to which is applied to the processed will be subtracted from the processed signal.  Or have I outwitted myself?  but in general, if we are talking about sound design, then such a utility is quite a cool thing in the context of my sound research.  I mean the ability to subtract from the main signal mixed from the side chain.

---

My reply:

Try setting `Side Mix` to 0. It might be the configuration you are looking for. Following is the mixing part in code.

https://github.com/ryukau/VSTPlugins/blob/43c743ba5377875d132d727e0f9336e666838b22/ImmediateSpacer/source/dsp/dspcore.cpp#L150-L151

**Reply by KlashKontakt** [reply]
sorry ! my faulte

**Comment by KlashKontakt** [comment]
everything works great! Ultimately, this is a complete solution. thanks again for your work

**Comment by KlashKontakt** [comment]
I'm not a designer at all, but I tried to make the interface in terms of convenience
I hope it's not too terrible =)) 

![INTRFC_PLG](https://user-images.githubusercontent.com/87890552/224660281-37e663fb-2fcc-40d9-83d9-8b11d6c68f1f.jpg)
![INTRFC_PLG_2](https://user-images.githubusercontent.com/87890552/224660298-a0a2a154-b8fc-4fd7-a9c2-a0fa24b66297.jpg)


have nice day

**Comment by ryukau** [comment]
@KlashKontakt Thanks for taking time for GUI. Is it working or just some photoshop?

Also could you provide the font you used in the image? I'd like to add a style similar to your image into this repository. It looks nice.

In case if the image is photoshop, then I have already made GUI as following. I should have communicated it earlier to avoid duplication of effort. Sorry about that.

![ImmediateSpacer](https://user-images.githubusercontent.com/6243462/224873408-68c38345-f7c8-4189-a24f-067ffb1c8aa0.png)

![ImmediateSpacer_SplashScreen](https://user-images.githubusercontent.com/6243462/224873423-c81dd61c-fbf1-4ffb-85c3-76746ce33a9d.png)

I'll probably change the name to RingModSpacer, but still thinking about it.

Have a nice day, too!

**Comment by KlashKontakt** [comment]
Go aisatsu Takamitsu!  (I hope I wrote it right)

font https://fontstruct.com/fontstructions/show/413734/stealth57

I made the interface in vector. here is the project in svg format. I hope this will be of some use to you
![INTRFC](https://user-images.githubusercontent.com/87890552/224900407-f4470ec2-dc77-4787-a3e8-c6649cd28d1c.svg)
The interface is my personal initiative and an attempt at a little participation.

The new name seems to me more understandable, since at first glance you can not immediately understand what it is for.

I can say that this tool works great and I don’t even know what. more can be added here. If only to offer some new ideas for new tools. what do you think about morphing 2 instruments? =) (of cause side chain connection)

https://youtu.be/nt9tXXaXRrM

Arigato and have nice day!

**Comment by ryukau** [comment]
Привет!

"Goaisatsu itasimasu" (ご挨拶いたします) might be used as in "greetings", but it's a bit too formal and I haven't used it in my life. "Konnichiwa" (こんにちは) is more common.

---

I made a style based on your image. It looks like following.

![キャプチャ](https://user-images.githubusercontent.com/6243462/225742746-d7e7fa24-18c4-40d8-b0b4-f2c040328c7b.PNG)

Following is the content of `style.json`. If you haven't, check out ["GUI Style Configuration" section on the manual](https://ryukau.github.io/VSTPlugins/manual/ParallelDetune/ParallelDetune_en.html#gui-style-configuration) to how to use it.

```json
{
  "fontFamily": "VT323",
  "fontBold": false,
  "fontItalic": false,
  "foreground": "#ffffff",
  "foregroundButtonOn": "#ffffff",
  "foregroundInactive": "#c0cad8",
  "background": "#8e99b9",
  "boxBackground": "#66748f",
  "border": "#66748f",
  "borderCheckbox": "#66748f",
  "borderLabel": "#66748f",
  "unfocused": "#dddddd",
  "highlightMain": "#c0cad8",
  "highlightAccent": "#3ecbd8",
  "highlightButton": "#ffb380",
  "highlightWarning": "#ffaaaa",
  "overlay": "#ffffffc0",
  "overlayHighlight": "#3ecbd840"
}
```

stealth57 font was too wide for my plugins, so I used VT323 font instead. See the link below.

- [VT323 - Google Fonts](https://fonts.google.com/specimen/VT323)

Thanks again for the image. I'm not good at graphic design. This kind of contribution helps a lot.

---

Morphing is not possible with my current skills. Morphing is difficult because it can easily become just another implementation of cross-fade. I have to put some research into it, but it's not priority for now.

**Comment by ryukau** [comment]
I released the idea discussed here as RingModSpacer. Download link is below.

https://github.com/ryukau/VSTPlugins/releases/tag/UhhyouPlugins0.54.0

I close this discussion as completed. If there's something to add, feel free to add comment or open another discussion.
